src/components/Calculator.test.js

```diff
@@ -1,10 +1,20 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { render, fireEvent } from '@testing-library/react';
 import Calculator from './Calculator';
 
 describe('Calculator', () => {
-  it('renders correctly', () => {
-    const { asFragment } = render(<Calculator />);
-    expect(asFragment()).toMatchSnapshot();
+  it('should render the component', () => {
+    const { container } = render(<Calculator />);
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('should update the screen on button click', () => {
+    const { getByTestId, getByText } = render(<Calculator />);
+    fireEvent.click(getByText('2'));
+    fireEvent.click(getByText('+'));
+    fireEvent.click(getByText('3'));
+    fireEvent.click(getByText('='));
+    expect(getByTestId('screen')).toHaveTextContent('5');
   });
```

Comment:
Great job @emhamza, I noticed that you implemented the use of React-testing-library in your code, which was a requirement requested by the previous code reviewer. Your implementation looks solid and demonstrates a good understanding of how to simulate user interactions and test the behavior of components using the testing library. 

By incorporating testing into your development workflow, you can catch potential bugs and regressions early on and ensure that your code is working as expected. Keep up the good work!